mmcls/structures/multi_task_data_sample.py

```diff
@@ -39,7 +39,14 @@ def set_gt_task(self, value: Dict) -> 'MultiTaskDataSample':
 
     def set_pred_task(self, value: Dict) -> 'MultiTaskDataSample':
         """Set score of ``pred_task``."""
-        self.pred_task = LabelData(**value)
+        if list(value.values())[0] == torch.tensor:
+            self.pred_task = LabelData(**value)
+        else:
```

Comment:
I would suggest a more strict approah here like 
```suggestion
        elif type(list(value.values())[0]) in self.data_samples_map.keys():
```

And raise an exception if the type is not known